### PR TITLE
Bugs fix

### PR DESCRIPTION
**Pull request by DmitryBogd**

### CODE REVIEW

**Review by JuliaNortman — APPROVED** [review]

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 1]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
     this.image = imageName;
   }
 
+  public category(){
+    this.accountService.getCategories()
+      .subscribe((categ) =>{
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:75]: don't forget to unsubscribe

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 2]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
...
+        this.responseCategory = categ;
+        console.log(this.responseCategory);
+        this.categoryName = this.responseCategory;
+      })
+  }
+
+  public firm(){
+    this.accountService.getFirm()
+      .subscribe((firm) =>{
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:84]: don't forget to unsubscribe

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 3]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
...
+        this.responseFirm = firm;
+        console.log(this.responseFirm);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:86]: don't commit debug logs

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 4]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
     this.image = imageName;
   }
 
+  public category(){
+    this.accountService.getCategories()
+      .subscribe((categ) =>{
+        this.responseCategory = categ;
+        console.log(this.responseCategory);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:77]: don't commit debug logs

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 5]
```diff
@@ -82,21 +109,22 @@ export class AddProductComponent {
   onSubmit(): void {
     this.submitted = true;
     if (this.form.invalid) {
+      console.log(this.form.value);
+      console.log("dont work");
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:113]: don't commit error logs. If you want to notify a user, make a notification on the site

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 6]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
...
+  public category(){
+    this.accountService.getCategories()
+      .subscribe((categ) =>{
+        this.responseCategory = categ;
+        console.log(this.responseCategory);
+        this.categoryName = this.responseCategory;
+      })
+  }
+
+  public firm(){
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:82]: why is this method public?

File: marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts [hunk 7]
```diff
@@ -62,6 +70,24 @@ export class AddProductComponent {
     this.image = imageName;
   }
 
+  public category(){
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/add-product/add-product.component.ts:73]: why is this method public?

File: marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts [hunk 8]
```diff
@@ -24,33 +24,59 @@ export class UpdateProductComponent implements OnInit{
...
   updated = false;
 
   response: any;
-
+  responseCategory: any;
+  responseFirm: any;
   image: string = '';
 
   ngOnInit(){
     //.subscribe((response) => {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts:43]: don't commit commented out code

File: marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts [hunk 9]
```diff
@@ -24,33 +24,59 @@ export class UpdateProductComponent implements OnInit{
...
-      this.accountService.getProductInfo(this.route.snapshot.params.id)
-        .subscribe((response) => {
-          this.response = response;
-          console.log(this.response);
-          this.formCreation();
-        })
+    this.accountService.getProductInfo(this.route.snapshot.params.id)
+      .subscribe((response) => {
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts:45]: need to unsubscribe

File: marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts [hunk 10]
```diff
@@ -24,33 +24,59 @@ export class UpdateProductComponent implements OnInit{
...
+        this.response = response;
+        console.log(this.response);
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts:47]: don't commit logs

File: marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts [hunk 11]
```diff
@@ -24,33 +24,59 @@ export class UpdateProductComponent implements OnInit{
...
+  }
+
+  public firm(){
+    this.accountService.getFirm()
+      .subscribe((firm) =>{
+        this.responseFirm = firm;
+        console.log(this.responseFirm);
+        this.firmName = this.responseFirm;
+      })
+  }
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts:78]: unsubscriptions, logs and unnecessary public access

File: marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts [hunk 12]
```diff
@@ -24,33 +24,59 @@ export class UpdateProductComponent implements OnInit{
...
   }
 
+  public category(){
+    this.accountService.getCategories()
+      .subscribe((categ) =>{
+        this.responseCategory = categ;
+        console.log(this.responseCategory);
+        this.categoryName = this.responseCategory;
+      })
+  }
```
Comment (MaxxMorozov) [inline marketplace-client/src/marketplace/src/app/product-catalog/update-product/update-product.component.ts:69]: unsubscriptions, logs and unnecessary public access

**Review by MaxxMorozov — COMMENTED** [review]
don't commit `node_modules`. Put them in the `git-ignore`

**Pull request merged**